rebound/librebound.py

```diff
@@ -215,7 +294,11 @@ def add(self, particle=None, **kwargs):
                 for p in particle:
                     self.add(p)
             elif isinstance(particle,str):
+                for key, unit in self.units.items(): # check if units aren't set to set defaults
```

Comment:
I would write this as 
   
    if None in self.units:
        self.units =  (.....)